libs/types-common/src/Data/Misc.hs

```diff
@@ -251,6 +268,9 @@ instance Cql HttpsUrl where
   fromCql (CqlBlob t) = runParser parser (toStrict t)
   fromCql _ = fail "HttpsUrl: Expected CqlBlob"
 
+instance Arbitrary HttpsUrl where
+  arbitrary = pure $ HttpsUrl [URI.QQ.uri|https://example.com|]
```

Comment:
I wrote this, I know, but why not:

```suggestion
  arbitrary = do
    host <- (<> ".") . ("host" <>) . pack . show . abs <$> (arbitrary @Int)
    pure 
      . HttpsUrl
      . (authorityL . _Just . authorityHostL . hostBSL %~ (host <>))
      $ [URI.QQ.uri|https://example.com|]
```

Reply:
Not sure how much better this is, as it still only covers trivial cases. I'd rather make that obvious by using a plain `pure x` instance. A proper instance would be even nicer of course, but I think that would be quite a bit of work.

Reply:
the benefit of having different ones is that it covers tests where we compare two urls that are supposed to be different --

ok, i just finished that thought, and it doesn't make sense anymore.  never mind.  :)